src/assets/TestFramework/ClientDiagnosticListener.cs

```diff
@@ -0,0 +1,264 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+
+namespace Azure.Core.Tests
+{
+    public class ClientDiagnosticListener : IObserver<KeyValuePair<string, object>>, IObserver<DiagnosticListener>, IDisposable
```

Comment:
@christothes, @JoshLove-msft, do you know whether we have anything like this in the test framework already?

Reply:
yes, that is actually from the Azure.Core. Let me update the copy script

Reply:
Oh awesome, thanks, @AlexanderSher!